src/BitChopp.Main/ViewModels/MainViewModel.cs

```diff
@@ -41,14 +46,27 @@ private async void OnSwitchSelected(SwitchCommandObject swObj)
         // Logic to handle switch selection, such as navigating to a new screen and displaying the QR code
         Console.WriteLine($"Selected Switch's Lnurl: {swObj.Switch.Lnurl}");
 
-        var qrWindow = new QRCodeWindow(DeviceId, swObj.Switch.Pin, swObj.Switch.Lnurl, _configService);
-        var task = qrWindow.ShowDialog(swObj.Window);
-        qrWindow.Topmost = true; // Make the window always on top
-        await task;
+        var qrWindow = new QRCodeWindow(DeviceId, swObj.Switch.Pin, swObj.Switch.Lnurl, _configService)
+        {
+            Topmost = true // Make the window always on top
+        };
+        await qrWindow.ShowDialog(swObj.Window);
+
+        // if (qrWindow.WebSocketResult == "Paid")
+        // {
+        await ShowSuccessWindow(swObj); // replace with actual value
+        // }
```

Comment:
should you uncomment the if?

Reply:
Just testing for now. When we have the proper way of testing without paying I'll remove it.

Reply:
@nGoline Maybe include an env var to skip the payment request or something like that?

`if (qrWindow.WebSocketResult == "Paid" || isDebugMode)`